File: Example/THLabelExample/TableViewController.m
//
//  TableViewController.m
//  THLabelExample
//
//  Created by Tobias Hagemann on 09/09/14.
//  Copyright (c) 2014 tobiha.de. All rights reserved.
//

#import "TableViewController.h"
#import "THLabel.h"

@interface TableViewCell : UITableViewCell
@property (nonatomic, weak) IBOutlet THLabel *customTextLabel;
@end

@implementation TableViewCell
@end

@implementation TableViewController

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
	return 100;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
	static NSString *CellIdentifier = @"Cell";
	TableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier forIndexPath:indexPath];
	cell.customTextLabel.textColor = [UIColor whiteColor];
	cell.customTextLabel.strokeColor = kStrokeColor;
	cell.customTextLabel.strokeSize = kStrokeSize;
	cell.customTextLabel.gradientStartColor = kGradientStartColor;
	cell.customTextLabel.gradientEndColor = kGradientEndColor;
	return cell;
}

@end

File: Example/THLabelExample/TableViewController.swift
//
//  TableViewController.swift
//  THLabelExample
//
//  Created by Vitalii Parovishnyk on 1/12/17.
//  Copyright © 2017 IGR Software. All rights reserved.
//

import UIKit

class TableViewCell: UITableViewCell {
    @IBOutlet weak fileprivate var customTextLabel: THLabel?
}

class TableViewController: UITableViewController {
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 100
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let CellIdentifier = "Cell"
        
        let cell = tableView.dequeueReusableCell(withIdentifier: CellIdentifier, for: indexPath) as! TableViewCell
        
        cell.customTextLabel?.textColor = UIColor.white;
        cell.customTextLabel?.strokeColor = Constants.kStrokeColor;
        cell.customTextLabel?.strokeSize = Constants.kStrokeSize;
        cell.customTextLabel?.gradientStartColor = Constants.kGradientStartColor;
        cell.customTextLabel?.gradientEndColor = Constants.kGradientEndColor;
        
        return cell;
    }

}
